binder/src/test/java/io/grpc/binder/SecurityPoliciesTest.java

```diff
@@ -164,11 +200,170 @@ public void testOneOfSignature_succeedsIfPackageNameAndOneOfSignaturesMatch()
   public void testHasSignature_failsIfUidUnknown() throws Exception {
     policy =
         SecurityPolicies.hasSignature(
-          packageManager,
-          appContext.getPackageName(),
-          SIG1);
+            packageManager,
+            appContext.getPackageName(),
+            SIG1);
 
     assertThat(policy.checkAuthorization(OTHER_UID_UNKNOWN).getCode())
-      .isEqualTo(Status.UNAUTHENTICATED.getCode());
+        .isEqualTo(Status.UNAUTHENTICATED.getCode());
+  }
+
+  @Test
+  public void testHasPermissions_sharedUserId_succeedsIfAllPackageHavePermissions()
+      throws Exception {
+    PackageInfo info =
+        newBuilder()
+            .setPackageName(OTHER_UID_PACKAGE_NAME)
+            .setPermission(ACCESS_FINE_LOCATION, REQUESTED_PERMISSION_GRANTED)
+            .setPermission(ACCESS_COARSE_LOCATION, REQUESTED_PERMISSION_GRANTED)
+            .build();
+
+    PackageInfo infoSamePerms =
+        newBuilder()
+            .setPackageName(OTHER_UID_SAME_SIGNATURE_PACKAGE_NAME)
+            .setPermission(ACCESS_FINE_LOCATION, REQUESTED_PERMISSION_GRANTED)
+            .setPermission(ACCESS_COARSE_LOCATION, REQUESTED_PERMISSION_GRANTED)
+            .build();
+
+    installPackages(OTHER_UID, info, infoSamePerms);
+
+    policy =
+        SecurityPolicies.hasPermissions(
+            packageManager, ImmutableSet.of(ACCESS_FINE_LOCATION, ACCESS_COARSE_LOCATION));
+    assertThat(policy.checkAuthorization(OTHER_UID).getCode()).isEqualTo(Status.OK.getCode());
+  }
+
+  @Test
+  public void testHasPermissions_sharedUserId_failsIfOnePackageHasNoPermissions() throws Exception {
+    PackageInfo info =
+        newBuilder()
+            .setPackageName(OTHER_UID_PACKAGE_NAME)
+            .setPermission(ACCESS_FINE_LOCATION, REQUESTED_PERMISSION_GRANTED)
+            .build();
+
+    PackageInfo infoNoPerms =
+        newBuilder()
+            .setPackageName(OTHER_UID_SAME_SIGNATURE_PACKAGE_NAME)
+            .setPermission(ACCESS_FINE_LOCATION, 0)
+            .build();
+
+    installPackages(OTHER_UID, info, infoNoPerms);
+
+    policy = SecurityPolicies.hasPermissions(packageManager, ImmutableSet.of(ACCESS_FINE_LOCATION));
+    assertThat(policy.checkAuthorization(OTHER_UID).getCode())
+        .isEqualTo(Status.PERMISSION_DENIED.getCode());
+    assertThat(policy.checkAuthorization(OTHER_UID).getDescription())
+        .contains("does not have permission");
```

Comment:
Can you change this to .contains(ACCESS_FINE_LOCATION) and add a .contains(OTHER_UID_SAME_SIGNATURE_PACKAGE_NAME) check ? That will ensure that the details needed for debugging SecurityPolicy failures are present in the description AND that this test is not accidentally failing for some unrelated setup reason.

Reply:
should be good now. modified all 3 applicable tests as well.